src/backend/Altinn.Receipt/Program.cs

```diff
@@ -89,30 +88,27 @@ async Task SetConfigurationProviders(ConfigurationManager config)
 
 async Task ConnectToKeyVaultAndSetApplicationInsights(ConfigurationManager config)
 {
-    KeyVaultSettings keyVaultSettings = new KeyVaultSettings();
+    KeyVaultSettings keyVaultSettings = new();
     config.GetSection("kvSetting").Bind(keyVaultSettings);
     if (!string.IsNullOrEmpty(keyVaultSettings.ClientId) &&
         !string.IsNullOrEmpty(keyVaultSettings.TenantId) &&
         !string.IsNullOrEmpty(keyVaultSettings.ClientSecret) &&
         !string.IsNullOrEmpty(keyVaultSettings.SecretUri))
     {
         logger.LogInformation("Program // Configure key vault client // App");
+        Environment.SetEnvironmentVariable("AZURE_CLIENT_ID", keyVaultSettings.ClientId);
+        Environment.SetEnvironmentVariable("AZURE_CLIENT_SECRET", keyVaultSettings.ClientSecret);
+        Environment.SetEnvironmentVariable("AZURE_TENANT_ID", keyVaultSettings.TenantId);
+        var azureCredentials = new DefaultAzureCredential();
+
+        config.AddAzureKeyVault(new Uri(keyVaultSettings.SecretUri), azureCredentials);
+
+        SecretClient client = new SecretClient(new Uri(keyVaultSettings.SecretUri), azureCredentials);
 
-        string connectionString = $"RunAs=App;AppId={keyVaultSettings.ClientId};" +
-                                  $"TenantId={keyVaultSettings.TenantId};" +
-                                  $"AppKey={keyVaultSettings.ClientSecret}";
-        AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(connectionString);
-        KeyVaultClient keyVaultClient = new KeyVaultClient(
-            new KeyVaultClient.AuthenticationCallback(
-                azureServiceTokenProvider.KeyVaultTokenCallback));
-        config.AddAzureKeyVault(
-            keyVaultSettings.SecretUri, keyVaultClient, new DefaultKeyVaultSecretManager());
         try
         {
-            SecretBundle secretBundle = await keyVaultClient
-                .GetSecretAsync(keyVaultSettings.SecretUri, vaultApplicationInsightsKey);
-
-            applicationInsightsConnectionString = string.Format("InstrumentationKey={0}", secretBundle.Value);
+            KeyVaultSecret keyVaultSecret = await client.GetSecretAsync(vaultApplicationInsightsKey);
+            applicationInsightsConnectionString = string.Format("InstrumentationKey={0}", keyVaultSecret.Value);
```

Comment:
In Storage you added a step. keyVaultSecret.Value.Value?

Reply:
Removed it again in Storage. Specifying the output as KeyVaultSecret  apparently changes the return type to be the secret directly so we don't need to get value twice